packages/rich-text-editor/src/RichTextEditor/RichTextEditor.spec.tsx

```diff
@@ -0,0 +1,167 @@
+  }
+
+  it("renders list buttons when receiving a list controls", () => {
+    render(<TestRTE {...defaultListArgs} />)
+
+    const bulletButton = screen.getByRole("button", { name: "Bullet List" })
+    const orderedButton = screen.getByRole("button", { name: "Numbered List" })
+
+    expect(bulletButton && orderedButton).toBeTruthy()
+  })
+
+  it("renders indent buttons when receiving a list controls", () => {
+    render(<TestRTE {...defaultListArgs} />)
+
+    const decreaseIndentBtn = screen.getByRole("button", {
+      name: "Decrease indent",
+    })
+    const increaseIndentBtn = screen.getByRole("button", {
+      name: "Increase indent",
+    })
+
+    expect(decreaseIndentBtn && increaseIndentBtn).toBeTruthy()
+  })
+
+  describe("Creating list nodes with buttons", () => {
+    it("will create a <ul> when user clicks the bullet list button", async () => {
+      render(<TestRTE {...defaultListArgs} />)
+
+      // We would use userEvent.type but contenteditable is not supported
+      // see thread: https://github.com/testing-library/user-event/issues/230
+      fireEvent.focus(screen.getByRole("textbox", { name: "List RTE" }), {
+        target: { textContent: "this will be a ul" },
+      })
+
+      await waitFor(() => {
+        userEvent.click(screen.getByRole("button", { name: "Bullet List" }))
+        screen.getByRole("list")
+
+        expect(screen.getByRole("list")).toBeInTheDocument()
+        expect(document.querySelectorAll("ul").length).toBeGreaterThan(0)
+      })
+    })
+
+    it("will create a <ol> when user clicks the numbered list button", async () => {
+      render(<TestRTE {...defaultListArgs} />)
+
+      fireEvent.focus(screen.getByRole("textbox", { name: "List RTE" }), {
+        target: { textContent: "this will be a ol" },
+      })
+
+      await waitFor(() => {
+        userEvent.click(screen.getByRole("button", { name: "Numbered List" }))
+
+        expect(screen.getByRole("list")).toBeInTheDocument()
+        expect(document.querySelectorAll("ol").length).toBeGreaterThan(0)
+      })
+    })
+  })
+
+  describe("Handling disabled indent states", () => {
+    const rteListData = [
+      {
+        type: "bulletList",
+        content: [
+          {
+            type: "listItem",
+            content: [
+              {
+                type: "paragraph",
+                content: [
+                  {
+                    type: "text",
+                    text: "First list item",
+                  },
+                ],
+              },
+            ],
+          },
+          {
+            type: "listItem",
+            content: [
+              {
+                type: "paragraph",
+                content: [
+                  {
+                    type: "text",
+                    text: "Second list item",
+                  },
+                ],
+              },
+            ],
+          },
+        ],
+      },
+    ]
+
+    it("indent buttons are 'disabled' by default", () => {
```

Comment:
/nit these test cases don't read nicely like your others "it indent buttons..."

Reply:
nit away :) could definitely update this!